app/retail/views.py

```diff
@@ -1089,17 +1089,53 @@ def results(request, keyword=None):
 def activity(request):
     """Render the Activity response."""
     page_size = 15
+    page = int(request.GET.get('page', 1))
+    what = request.GET.get('what', 'everywhere')
+
+    # create diff filters
     activities = Activity.objects.all().order_by('-created_on')
+    if request.user.is_authenticated:
+        relevant_profiles = []
+        if what == 'tribes':
+            from dashboard.models import get_my_earnings
```

Comment:
seems like this will get called alot, so we should definitely make the effort to move this import if possible

Reply:
done